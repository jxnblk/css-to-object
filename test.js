const test = require('ava')
const toObj = require('./index')

test('returns an object', t => {
  const obj = toObj(`color: tomato`)
  t.is(typeof obj, 'object')
})

test('parses media queries', t => {
  const obj = toObj(`
    color: tomato;
    margin-bottom: 16px;
    @media (min-width: 40em) {
      color: green;
    }
  `, {
    numbers: true,
    camelCase: true,
  })
  t.is(typeof obj, 'object')
  t.deepEqual(obj, {
    color: 'tomato',
    marginBottom: 16,
    '@media (min-width: 40em)': {
      color: 'green'
    }
  })
})

test('parses psuedoclasses', t => {
  const obj = toObj(`
    color: tomato;
    &:hover {
      color: green;
    }
  `)
  t.deepEqual(obj, {
    color: 'tomato',
    ':hover': {
      color: 'green'
    }
  })
})

test('parses psuedoclasses with option to preserve ampersands (&)', t => {
  const obj = toObj(`
    color: tomato;
    &:hover {
      color: green;
    }
    & span {
      color: blue;
    }
  `, {
    ampersands: true
  })
  t.deepEqual(obj, {
    color: 'tomato',
    '&:hover': {
      color: 'green'
    },
    '& span': {
      color: 'blue'
    }
  })
})

test('parses nested selectors', t => {
  const obj = toObj(`
    color: tomato;
    &:hover {
      color: green;
    }
    & span {
      color: blue;
    }
  `)
  t.deepEqual(obj, {
    color: 'tomato',
    ':hover': {
      color: 'green'
    },
    'span': {
      color: 'blue'
    }
  })
})

test('snapshot', t => {
  const obj = toObj(`
    color: tomato;
    margin: 32px;
    padding: 6px 9px;
    left: -1px;
    @media (min-width: 40em) {
      margin: 48px;
    }
    &:hover {
      color: black;
    }
    h1 {
      font-size: 32px;
    }
  `, {
    camelCase: true,
    numbers: true
  })
  t.snapshot(obj)
})
