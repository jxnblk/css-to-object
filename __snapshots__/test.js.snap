// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  ":hover": Object {
    "color": "black",
  },
  "@media (min-width: 40em)": Object {
    "margin": 48,
  },
  "color": "tomato",
  "h1": Object {
    "fontSize": 32,
  },
  "left": -1,
  "margin": 32,
  "padding": "6px 9px",
}
`;
